domain/consensus/consensus.go

```diff
@@ -112,6 +112,30 @@ func (s *consensus) GetBlock(blockHash *externalapi.DomainHash) (*externalapi.Do
 	return block, nil
 }
 
+func (s *consensus) GetBlockEvenIfHeaderOnly(blockHash *externalapi.DomainHash) (*externalapi.DomainBlock, error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	stagingArea := model.NewStagingArea()
+
+	block, err := s.blockStore.Block(s.databaseContext, stagingArea, blockHash)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			header, err := s.blockHeaderStore.BlockHeader(s.databaseContext, stagingArea, blockHash)
+			if err != nil {
+				if errors.Is(err, database.ErrNotFound) {
```

Comment:
Instead of building a pyramid, please exit out of the function when `err` is not ErrNotFound.